src/goto-programs/string_refine_preprocess.cpp

```diff
@@ -0,0 +1,1343 @@
+
+        auto it=string_functions.find(function_id);
+        if(it!=string_functions.end())
+          make_string_function(
+            goto_program, i_it, it->second, signature, false, false);
+
+        it=side_effect_functions.find(function_id);
+        if(it!=side_effect_functions.end())
+          make_string_function_side_effect(
+            goto_program, i_it, it->second, signature);
+
+        it=string_function_calls.find(function_id);
+        if(it!=string_function_calls.end())
+          make_string_function_call(goto_program, i_it, it->second, signature);
+
+        it=string_of_char_array_functions.find(function_id);
+        if(it!=string_of_char_array_functions.end())
+          make_char_array_function(
+            goto_program, i_it, it->second, signature, 0);
+
+        it=string_of_char_array_function_calls.find(function_id);
+        if(it!=string_of_char_array_function_calls.end())
+          make_char_array_function_call(
+            goto_program, i_it, it->second, signature);
+
+        it=side_effect_char_array_functions.find(function_id);
+        if(it!=side_effect_char_array_functions.end())
+          make_char_array_side_effect(
+            goto_program, i_it, it->second, signature);
+
+        if(function_id==irep_idt("java::java.lang.String.toCharArray:()[C"))
+          make_to_char_array_function(goto_program, i_it);
+      }
+    }
+    else
+    {
+      if(i_it->is_assign())
+      {
+        // In assignments we replace string literals and C string functions
+        code_assignt assignment=to_code_assign(i_it->code);
+
+        exprt new_rhs=assignment.rhs();
+        code_assignt new_assignment(assignment.lhs(), new_rhs);
+
+        if(new_rhs.id()==ID_function_application)
+        {
+          function_application_exprt f=to_function_application_expr(new_rhs);
+          const exprt &name=f.function();
+          assert(name.id()==ID_symbol);
+          const irep_idt &id=to_symbol_expr(name).get_identifier();
+          auto it=c_string_functions.find(id);
+          if(it!=c_string_functions.end())
+          {
+            declare_function(it->second, f.type());
+            f.function()=symbol_exprt(it->second);
+            new_assignment=code_assignt(assignment.lhs(), f);
+          }
+        }
+
+        new_assignment.add_source_location()=assignment.source_location();
+        i_it->make_assignment();
+        i_it->code=new_assignment;
+      }
+    }
+  }
+  return;
+}
+
+/*******************************************************************\
+
+Function: string_refine_preprocesst::initialize_string_function_table
+
+ Purpose: fill maps with correspondance from java method names to cprover
+          functions
+
+\*******************************************************************/
+
+void string_refine_preprocesst::initialize_string_function_table()
+{
+  string_functions
+    [irep_idt("java::java.lang.String.codePointAt:(I)I")]=
+    ID_cprover_string_code_point_at_func;
```

Comment:
With C++11 you should be able to initialize that map using { { id, value}, {id, value} , ... } as a more concise syntax.